### PR TITLE
Follow up design tweaks for global styles presets

### LINKED ISSUE #59948: Global styles: Active color variation outline is inverted
### Description

The currently selected color variation is showing as not having an outline, the opposite as what the typography variation does.

### Step-by-step reproduction instructions

Using TT4, go to Site editor > Styles

Select a color variation, check the outline of all the options

### Screenshots, screen recording, code snippet

Screenshot 2024-03-18 at 12 02 07

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by richtabor**
## What?

A follow-up to https://github.com/WordPress/gutenberg/pull/56622, in support of https://github.com/WordPress/gutenberg/issues/56604.

Fixes https://github.com/WordPress/gutenberg/issues/59948

## Testing Instructions

1. Open the site editor.
2. Open Styles.
3. Select "Colors". 
4. See changes.
5. Select "Typography".
6. See changes.

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
|CleanShot 2024-03-19 at 14 48 15|CleanShot 2024-03-19 at 14 52 03|

### CODE REVIEW

File: packages/edit-site/src/components/global-styles/variations/style.scss [hunk 1]
```diff
@@ -5,34 +5,45 @@
...
-		box-shadow: 0 0 0 $border-width $gray-200;
 		// Shown in Windows 10 high contrast mode.
 		outline: 1px solid transparent;
+		position: relative;
+		transition: box-shadow 0.1s linear;
+
+		&::after {
+			content: "";
+			position: absolute;
+			top: -1px;
```
Comment (jasmussen) [inline packages/edit-site/src/components/global-styles/variations/style.scss]: You can use $border-width instead of 1px. Mainly for legibility.

**Review by jasmussen — APPROVED** [review]

File: packages/edit-site/src/components/global-styles/variations/style.scss (hunk 1 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/variations/style.scss]: Good look; added via https://github.com/WordPress/gutenberg/pull/60031/commits/1910ef887a1e66e8852a5ca21618acb7bc7865fd.

File: packages/edit-site/src/components/global-styles/variations/style.scss [hunk 2]
```diff
@@ -5,34 +5,45 @@
...
 	.edit-site-global-styles-variations_item-preview {
-		padding: $border-width * 2;
 		border-radius: $radius-block-ui;
-		box-shadow: 0 0 0 $border-width $gray-200;
 		// Shown in Windows 10 high contrast mode.
 		outline: 1px solid transparent;
+		position: relative;
+		transition: box-shadow 0.1s linear;
+
+		&::after {
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/variations/style.scss:16]: Should this be a mixin?

**Comment by scruffian** [comment]
This readds the "Presets" heading, I assume that is intentional?

File: packages/edit-site/src/components/global-styles/font-families.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ function FontFamilies() {
 				/>
 			) }
 
-			<VStack spacing={ 3 }>
+			<VStack spacing={ 2 }>
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/font-families.js:40]: This is wrapped in another VStack inside screen-typography. Maybe we should just remove that one and add that space to this one?

File: packages/edit-site/src/components/global-styles/style.scss [hunk 4]
```diff
@@ -53,15 +53,29 @@
...
 		border-top: none;
+		row-gap: calc(#{$grid-unit-05} * 3);
 	}
 }
 
+.edit-site-global-styles-screen-colors__palette-count {
+	color: $gray-700;
+}
+
+.edit-site-global-styles-sidebar__more-menu > .components-button.has-icon {
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/style.scss]: What element this is targetting?

**Comment by sahiladit** [comment]
Hey can you help me ,how you did this?How do you created a pull req?

File: packages/edit-site/src/components/global-styles/style.scss [hunk 5]
```diff
@@ -53,15 +53,29 @@
 }
 
 .edit-site-global-styles-screen-colors {
-	margin: $grid-unit-20;
+	margin: $grid-unit-10 $grid-unit-20 $grid-unit-20;
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/style.scss]: This no longer matches `.edit-site-global-styles-screen-typography`. We should probably combine into a new shared class.

**Comment by scruffian** [comment]
> Hey can you help me ,how you did this?How do you created a pull req?

You need to fork the repo, and then create a new branch on your fork. There's a lot more detail here: https://github.com/WordPress/gutenberg/blob/c1032abaf71e86f0effe13a181ba189c083f98ef/docs/contributors/code/README.md

File: packages/edit-site/src/components/global-styles/variations/variations-typography.js [hunk 6]
```diff
@@ -14,8 +14,9 @@ import { useTypographyVariations } from '../hooks';
 import TypographyExample from '../typography-example';
 import PreviewIframe from '../preview-iframe';
 import Variation from './variation';
+import Subtitle from '../subtitle';
 
-export default function TypographyVariations() {
+export default function TypographyVariations( { title, gap = 2 } ) {
```
Comment (scruffian) [inline packages/edit-site/src/components/global-styles/variations/variations-typography.js:19]: I'm not super keen on adding gap as a prop. Style variations have a gap of 3 in both contexts, so I'd rather keep that for consistency and simplicity.

File: packages/edit-site/src/components/global-styles/variations/style.scss (hunk 2 above)
Comment (scruffian) [reply packages/edit-site/src/components/global-styles/variations/style.scss:16]: If not then at least I think we should have a comment that explains what it does.

File: packages/edit-site/src/components/global-styles/variations/variations-typography.js (hunk 6 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/variations/variations-typography.js:19]: > Style variations have a gap of 3 in both contexts, so I'd rather keep that for consistency and simplicity.

They shouldn't though. In the smaller context of the Inspector, the gap should be `2`, but in the wider context of the dark material, they need a gap of `3`.

Comment (richtabor) [reply packages/edit-site/src/components/global-styles/variations/variations-typography.js:19]: If you have another idea to achieve the same result, I'm open to it.

File: packages/edit-site/src/components/global-styles/style.scss (hunk 5 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/style.scss]: I updated in https://github.com/WordPress/gutenberg/pull/60031/commits/775cca4aa70b588260fb6ed9caa5324f8f3168dc; there are a few nuances between them that I didn't want to dive too deep into. Perhaps worth a follow-up refactor, now that these two panels are coming closer together.

File: packages/edit-site/src/components/global-styles/style.scss (hunk 4 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/style.scss]: Ah, leftover from another exploration — removed in https://github.com/WordPress/gutenberg/pull/60031/commits/871883951e1544009d7a940a7b785dec6cb4e38c

File: packages/edit-site/src/components/global-styles/font-families.js (hunk 3 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/font-families.js:40]: Good catch; https://github.com/WordPress/gutenberg/pull/60031/commits/a668bb745982c3a6a96e5b58fc96b89a216a2b9a

**Comment by richtabor** [comment]
> This readds the "Presets" heading, I assume that is intentional?

Yes. After testing without it I think it's fine to include. Perhaps we can come up with a better solution and revisit, but for now this puts us in a good spot. 

I had this in mind, which works ok for colors—but not so much for typography. 

CleanShot 2024-03-20 at 14 24 25

**Comment by richtabor** [comment]
> This readds the "Presets" heading, I assume that is intentional?

Yes. After testing without it I think it's fine to include. Perhaps we can come up with a better solution and revisit, but for now this puts us in a good spot. 

I had this in mind, which works ok for colors—but not so much for typography. 

CleanShot 2024-03-20 at 14 24 25

File: packages/edit-site/src/components/global-styles/variations/style.scss (hunk 2 above)
Comment (richtabor) [reply packages/edit-site/src/components/global-styles/variations/style.scss:16]: We don't currently have a mixin for this. I did add a comment and the reduce-motion mixin for these: https://github.com/WordPress/gutenberg/pull/60031/commits/1fdbbeca8eb22ade40f560fcc822ca41f6ec8fa4

File: packages/edit-site/src/components/global-styles/variations/variations-typography.js (hunk 6 above)
Comment (scruffian) [reply packages/edit-site/src/components/global-styles/variations/variations-typography.js:19]: I updated the styles tiles to use a gap of 3 in site view.

**Review by scruffian — APPROVED** [review]
This looks good to me. Let's bring it in.

**Pull request merged**